java/core/src/test/java/sleeper/core/statestore/inmemory/InMemoryFileReferenceStore.java

```diff
@@ -144,6 +145,9 @@ public void atomicallyUpdateFilesToReadyForGCAndCreateNewActiveFiles(String jobI
             if (!jobId.equals(reference.getJobId())) {
                 throw new StateStoreException("File reference not assigned to job: " + jobId);
             }
+            if (newFilenames.contains(filename)) {
+                throw new StateStoreException("File reference has same filename as new file: " + filename);
+            }
```

Comment:
This new behaviour needs a test. That can be added for all the file reference stores.

Reply:
Thanks, added a test to all 3 statestores